src/webgpu/api/operation/adapter/requestDevice.spec.ts

```diff
@@ -1,16 +1,106 @@
 export const description = `
 Tests for GPUAdapter.requestDevice.
 
-TODO:
-- descriptor is {null, undefined}
-- descriptor with the combinations of features
+Test all possible options to requestAdapter.
+
+default, low-power, and high performance should all always return adapters
```

Comment:
I think that an implementation is technically allowed to return null here as well.

"If the user agent chooses to return an adapter, it should: Otherwise.. resolve  null"